components/pd_client/src/client.rs

```diff
@@ -98,6 +105,7 @@ impl RpcClient {
                         }
                     };
 
+                    // FIXME: RwLock may block the async executor.
```

Comment:
??

Reply:
if a lock is held across `.await` it will cause deadlock. the blocking behavior may also reduce the thread pool capacity if the `.wl()` is held for too long.

if the lock is released very quickly then it is not a problem and we can remove the FIXME (and add a comment to the RwLock explaining why it is fine).